src/libs/geolocation/GeolocationWrapper.tsx

```diff
@@ -17,70 +17,81 @@ type RequestGeolocPermissionParams = {
 
 export interface IGeolocationContext {
   position: GeoCoordinates | null
-  setPosition: (position: GeoCoordinates | null) => void
-  permissionState: GeolocPermissionState
+  appGeolocPermission: boolean
+  phoneSettingsGeolocPermission: GeolocPermissionState
+  setAppGeolocPermission: (newPermission: boolean) => void
   requestGeolocPermission: (params?: RequestGeolocPermissionParams) => Promise<void>
-  checkGeolocPermission: () => Promise<void>
-  triggerPositionUpdate: () => void
 }
 
 export const GeolocationContext = React.createContext<IGeolocationContext>({
   position: null,
-  setPosition: () => undefined,
-  permissionState: GeolocPermissionState.DENIED,
-  requestGeolocPermission: async () => {
+  appGeolocPermission: false,
+  phoneSettingsGeolocPermission: GeolocPermissionState.DENIED,
+  setAppGeolocPermission() {
     // nothing
   },
-  checkGeolocPermission: async () => {
+  async requestGeolocPermission() {
     // nothing
   },
-  triggerPositionUpdate: () => undefined,
 })
 
+async function getInitialAppGeolocPermission() {
+  const hasAllowedGeolocationInPhoneSettings: GeolocPermissionState = await checkGeolocPermission()
+  if (hasAllowedGeolocationInPhoneSettings !== GeolocPermissionState.GRANTED) {
+    return false
+  }
+  const hasAllowedGeolocationInApp = await storage.readObject<boolean>('has_allowed_geolocation')
+  if (!hasAllowedGeolocationInApp) {
+    return false
+  }
+  return true
+}
+
 export const GeolocationWrapper = ({ children }: { children: Element }) => {
   const [position, setPosition] = useState<GeoCoordinates | null>(null)
-  const [permissionState, setPermissionState] = useState<GeolocPermissionState>(
-    GeolocPermissionState.DENIED
-  )
-  const [shouldComputePosition, setShouldComputePosition] = useState(0)
+
+  const [appGeolocPermission, setAppGeolocPermission] = useState<boolean>(true)
+  const [phoneSettingsGeolocPermission, _setPhoneSettingsGeolocPermission] = useState<
```

Comment:
à quoi sert d'avoir `phoneSettingsGeolocPermission` dans le state, j'ai l'impression que c'est toujours consistant ave le résultat de `await requestGeolocPermission()` ?

Reply:
Quand `phoneSettingsGeolocPermission` est modifié, on a envie que React le détecte (en faisant la comparaison Object.is) et que le wrapper rerender les children et execute ses différents effets.